### PR TITLE
Translating 'reference-react.md'

### PR DESCRIPTION
**Pull request by hgareeballa**

### CODE REVIEW

**Comment by hgareeballa** [comment]
i have made changes to the file now, and its fully translated , please check

File: content/docs/reference-react.md [hunk 1]
```diff
@@ -212,131 +217,148 @@ You will not typically invoke `React.createFactory()` directly if you are using
...
+
+
+* [تمرير المراجع إلى مكوّنات  DOM .](/docs/forwarding-refs.html#forwarding-refs-to-dom-components)
+* [تمرير المراجع في المكوّنات ذات الترتيب الأعلى.](/docs/forwarding-refs.html#forwarding-refs-in-higher-order-components)
 
-* [Forwarding refs to DOM components](/docs/forwarding-refs.html#forwarding-refs-to-dom-components)
-* [Forwarding refs in higher-order-components](/docs/forwarding-refs.html#forwarding-refs-in-higher-order-components)
 
-`React.forwardRef` accepts a rendering function as an argument. React will call this function with `props` and `ref` as two arguments. This function should return a React node.
+قبل `React.forwardRef` دالة تصيير كوسيط لها. ستستدعي React هذه الدالة مع الخاصيّات `props` والمرجع `ref` كوسيطين لها. يجب أن تُعيد هذه الدالة عقدة React:
```
Comment (iRayan7) [inline content/docs/reference-react.md]: قبل -> تقبل

File: content/docs/reference-react.md [hunk 2]
```diff
@@ -351,10 +373,10 @@ function MyComponent() {
 }
 ```
 
-It is documented in our [code splitting guide](/docs/code-splitting.html#reactlazy). Note that `lazy` components can be deep inside the `Suspense` tree -- it doesn't have to wrap every one of them. The best practice is to place `<Suspense>` where you want to see a loading indicator, but to use `lazy()` wherever you want to do code splitting.
+جرى توثيق  في [صفحة تقسيم الشيفرة](/docs/code-splitting.html#reactlazy). انتبه إلى أنَّ المكونات `lazy` (الكسولة) يمكن أن تتوضع بداخل الشجرة `Suspense` بعمق، إذ لا تحتاج إلى تغليف كل واحدة منها. أفضل سلوك هو وضع `<Suspense>` حيث أردت رؤية مؤشر تحميل، ولكن استعمال `()lazy` حيثما أردت القيام بتقسيم الشيفرة.
 
-While this is not supported today, in the future we plan to let `Suspense` handle more scenarios such as data fetching. You can read about this in [our roadmap](/blog/2018/11/27/react-16-roadmap.html).
+لمَّا كان ذلك غير مدعوم الآن، فنخطط في المستقبل بترك `Suspense` يعالج حالات أوسع مثل جلب بيانات. يمكنك قراءة المزيد في هذه [التدوينة](/blog/2018/11/27/react-16-roadmap.html).
```
Comment (iRayan7) [inline content/docs/reference-react.md]: لمَّا كان ذلك -> طالما ان ذلك

**Comment by iRayan7** [comment]
🎉

**Pull request merged**